strawberry/relay/types.py

```diff
@@ -426,6 +397,44 @@ def _id(cls, root: Node, info: Info) -> GlobalID:
         # If node_id is not str, GlobalID will raise an error for us
         return GlobalID(type_name=type_name, node_id=str(node_id))
 
+    @classmethod
+    def resolve_id_attr(cls) -> str:
+        if cls._id_attr is None:
+            candidates: list[str] = []
+            for base in cls.__mro__:
+                base_namespace = sys.modules[base.__module__].__dict__
+
+                for attr_name, attr in getattr(base, "__annotations__", {}).items():
```

Comment:
wouldn't it be better to do it like in:
types/type_resolver/_get_fields
(use dataclasses.fields)

Reply:
In theory (and it was actually my first idea as well =P), but the `NodeID` is a `Private` field, meaning it will be skipped and not added to the type's fields